c++/src/kj/compat/tls.c++

```diff
@@ -312,78 +317,55 @@ private:
   ReadyInputStreamWrapper readBuffer;
   ReadyOutputStreamWrapper writeBuffer;
 
-  kj::Promise<size_t> tryReadInternal(
-      void* buffer, size_t minBytes, size_t maxBytes, size_t alreadyDone) {
-    return sslCall([this,buffer,maxBytes]() { return SSL_read(ssl, buffer, maxBytes); })
-        .then([this,buffer,minBytes,maxBytes,alreadyDone](size_t n) -> kj::Promise<size_t> {
-      if (n >= minBytes || n == 0) {
-        return alreadyDone + n;
-      } else {
-        return tryReadInternal(reinterpret_cast<byte*>(buffer) + n,
-                               minBytes - n, maxBytes - n, alreadyDone + n);
-      }
-    });
-  }
-
-  Promise<void> writeInternal(kj::ArrayPtr<const byte> first,
-                              kj::ArrayPtr<const kj::ArrayPtr<const byte>> rest) {
-    KJ_REQUIRE(shutdownTask == kj::none, "already called shutdownWrite()");
-
-    // SSL_write() with a zero-sized input returns 0, but a 0 return is documented as indicating
-    // an error. So, we need to avoid zero-sized writes entirely.
-    while (first.size() == 0) {
-      if (rest.size() == 0) {
-        return kj::READY_NOW;
+  Promise<void> writeInternal(kj::ArrayPtr<const kj::ArrayPtr<const byte>> slices) {
+    auto cork = writeBuffer.cork();
+    for (auto slice: slices) {
+      KJ_REQUIRE(shutdownTask == kj::none, "already called shutdownWrite()");
```

Comment:
If I'm understanding correctly, this would still differ in behavior from the original code, since it just checks for the presence of shutdownTask at the beginning of each slice, while the original code would effectively check before every call to `SSL_write()`, potentially several times per slice.

If we wanted to approximate the original behavior, I think we need to do the check at least at the beginning of the inner `while` loop.  If we want to exactly replicate the original behavior, we might also need to call it elsewhere with more complex logic, since it looks like it will also perform the check in situations where the list is empty or contains empty slices.